pkg/provider/resource_cluster.go

```diff
@@ -597,7 +638,63 @@ func (c *clusterResource) ensureClusterIsHealthy(ctx context.Context, cluster Cl
 		})
 }
 
-func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster {
+func (c *clusterResource) makeClusterForCreate(ctx context.Context, clusterResource ClusterResourceModel) (models.Cluster, error) {
+	cluster := generateGenericClusterModel(clusterResource)
+	// add BAH Code
+	if strings.Contains(clusterResource.CloudProvider.ValueString(), "bah") {
+		return c.addBAHFields(ctx, cluster, clusterResource)
+	} else {
+		return cluster, nil
+	}
+}
+
+func (c *clusterResource) addBAHFields(ctx context.Context, cluster models.Cluster, clusterResource ClusterResourceModel) (models.Cluster, error) {
+	clusterRscCSP := clusterResource.CloudProvider
+	clusterRscPrincipalIds := clusterResource.PeAllowedPrincipalIds
+	clusterRscSvcAcntIds := clusterResource.ServiceAccountIds
+
+	// If there is an existing Principal Account Id for that Region, use that one.
+	pids, err := c.client.GetPeAllowedPrincipalIds(ctx, clusterResource.ProjectId, clusterRscCSP.ValueString(), clusterResource.Region.ValueString())
+	if err != nil {
+		return cluster, err
```

Comment:
This needs to return empty cluster
```suggestion
		return models.Cluster{}, err
```